ShopperInsights/src/main/java/com/braintreepayments/api/ShopperInsightsClient.kt

```diff
@@ -91,16 +85,25 @@ class ShopperInsightsClient @VisibleForTesting internal constructor(
                 accountDetails = includeAccountDetails,
                 constraintType = constraintType,
                 paymentSources = paymentSources
-            )
-        )
-        // Hardcoded result
-        callback.onResult(
-            ShopperInsightsResult.Success(
-                ShopperInsightsInfo(
-                    isPayPalRecommended = isPaymentRecommended(result.eligibleMethods.paypal),
-                    isVenmoRecommended = isPaymentRecommended(result.eligibleMethods.venmo)
-                )
-            )
+            ),
+            callback = { result, error ->
+                if (error == null) {
+                    callback.onResult(
+                        ShopperInsightsResult.Success(
+                            ShopperInsightsInfo(
+                                isPayPalRecommended = isPaymentRecommended(result?.eligibleMethods?.paypal),
+                                isVenmoRecommended = isPaymentRecommended(result?.eligibleMethods?.venmo)
```

Comment:
If the `result` comes back `null`/missing, would we instead want to surface an error to the merchant, versus giving them back a `false` for `is<Payment>Recommended`?

Reply:
I'm adding this logic now - do we expect to always get Venmo and PayPal in the response? Are there any cases where we'd only get one or the other?

```kotlin
            result?.eligibleMethods?.paypal == null ||
                    result.eligibleMethods.venmo == null -> {
                callback.onResult(
                    ShopperInsightsResult.Failure(
                        NullPointerException("Missing data in API response")
                    )
                )
            }
```

This is what I currently have, if either of methods is missing I'm invoking the failure.

Reply:
i think based on the [doc](https://paypal.atlassian.net/wiki/spaces/MI/pages/738195099/Internal+Shopper+Insights+-+BT+Web+SDK+Integration) here we can either get both, either paypal or venmo, or neither. Thinking of throwing a custom error like `BTShopperInsightsError.emptyBodyReturned` in the case none are eligible. wdyt?

Reply:
thanks for sharing that doc!

if we're able to get one method and not the other, we might want to only have a check to see if both Venmo and PayPal are missing from the response and return the failure. 

If we get one and not the other, we could treat it as a successful onResult, but default the one that's missing to not recommended.

in the case that both are returned in the API but both are not eligible, it might make sense to return a success to the merchant but have both `isPayPalRecommended` and `isVenmoRecommended` as false. 

lots of edge cases here to handle 😅

Reply:
i just pushed a commit updating this logic to && instead of ||:

```kotlin
            result?.eligibleMethods?.paypal == null &&
                    result.eligibleMethods.venmo == null -> {
                callback.onResult(
                    ShopperInsightsResult.Failure(
                        NullPointerException("Missing data in API response")
                    )
                )
            }
```

what do you all think? does this logic make sense?

Reply:
Nit: I wouldn't use `NullPointerException`, but a more specific exception for the use-case. Otherwise, I'm good with the logic.

Reply:
the logic looks good to me! i tried to match the same on iOS and am throwing a custom error -`BTShopperInsightsError.emptyBodyReturned` - in this case.

Reply:
I'll update this to a specific exception matching `emptyBodyReturned` on iOS